src/main/java/hudson/plugins/parameterizedtrigger/BuildTriggerConfig.java

```diff
@@ -517,20 +526,111 @@ protected Future schedule(AbstractBuild<?, ?> build, final Job project, int quie
 
         // Includes both traditional projects via AbstractProject and Workflow Job
         if (project instanceof ParameterizedJobMixIn.ParameterizedJob) {
-            final ParameterizedJobMixIn<?, ?> parameterizedJobMixIn = new ParameterizedJobMixIn() {
+            final ParameterizedJobMixIn<?, ? extends AbstractBuild> parameterizedJobMixIn = new ParameterizedJobMixIn() {
                 @Override
-                protected Job<?, ?> asJob() {
+                protected Job<?, ? extends AbstractBuild> asJob() {
                     return project;
                 }
             };
+            
+            QueueTaskFuture<? extends AbstractBuild> existingBuild = connectToExistingBuild(build, project, queueActions);
+            if (existingBuild != null)
+            {
+                return existingBuild;
+            }
             return parameterizedJobMixIn.scheduleBuild2(quietPeriod, queueActions.toArray(new Action[queueActions.size()]));
         }
 
         // Trigger is not compatible with un-parameterized jobs
         return null;
     }
 
-    protected Future schedule(AbstractBuild<?, ?> build, Job project, List<Action> list) throws InterruptedException, IOException {
+    private QueueTaskFuture<? extends AbstractBuild> connectToExistingBuild(AbstractBuild<?, ?> currentBuild, Job<?, ? extends AbstractBuild> project, List<Action> actions) {
+        Run upstreamRoot = findUpstreamRoot(currentBuild, actions);
+
+        //always trigger the build if a user started the root build
+        Cause userIdCause = upstreamRoot.getCause(UserIdCause.class);
```

Comment:
@kwhetstone 
I need to remove this stuff. It's fits better to a new feature/pull request.
Is there any reason why the parameterized trigger plugin doesn't connect to jobs already build. My idea here is to over an option to allow using build result of job already finished and not trigger them again.

The scenario is as following: Having job A and B both depending on C. If C was build with the same parameters before and the result doesn't change if the parameters are the same (thats the tricky part), there is no need to rebuild it. There might be exceptional cases where a user might want to enforce building, so my first idea is to only reuse the result from a build finished in the past, if the build request wasn't started by a user (need to check the upper most upstream job). I think there should be some options (reduce number of builds by using results of existing build with same arguments + run a new build if uppermost upstream job was triggered by user) to keep compatibility and not breaking stuff for other users.